Sources/xcresultparser/SonarCoverageConverter.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  SonarCoverageConverter.swift
+//  xcresult2text
+//
+//  Created by Alex da Franca on 11.06.21.
+//
+
+import Foundation
+import XCResultKit
+
+public class SonarCoverageConverter: CoverageConverter {
+    public override func xmlString(quiet: Bool) throws -> String {
+        let coverageXML = XMLElement(name: "coverage")
+        coverageXML.addAttribute(name: "version", stringValue: "1")
+        let coverageXMLSemaphore = DispatchSemaphore(value: 1)
+        let files = try coverageFileList()
+        
+        // since we need to invoke xccov for each file, it takes pretty much time
+        // so we invoke it in parallel on 8 threads, that speeds up things considerably
+        let queue = OperationQueue()
+        queue.maxConcurrentOperationCount = 8 //Deadlock if this is = 1
+        queue.qualityOfService = .userInitiated
+        for file in files {
+            guard !file.isEmpty else { continue }
+            if !quiet {
+                writeToStdError("Coverage for: \(file)\n")
+            }
+            let op = BlockOperation { [self] in
+                do {
+                    let coverage = try fileCoverageXML(for: file, relativeTo: projectRoot)
+                    coverageXMLSemaphore.wait()
+                    coverageXML.addChild(coverage)
+                    coverageXMLSemaphore.signal()
+                } catch {
+                    writeToStdErrorLn(error.localizedDescription)
+                }
+            }
+            queue.addOperation(op)
+        }
+        // This will block until all our operation have compleated (or been canceled)
+        queue.waitUntilAllOperationsAreFinished()
+        return coverageXML.xmlString(options: [.nodePrettyPrint, .nodeCompactEmptyElement])
+    }
+    
+    private func fileCoverageXML(for file: String, relativeTo projectRoot: String) throws -> XMLElement {
```

Comment:
It would be possible to DRY this out more by moving `fileCoverage` up to the superclass and having it return `FileInfo` structs, and then doing a second pass in the subclass to convert those to XML. Not sure if that's desired over how I have it currently implemented.

Reply:
Sounds good to me. I'll make the changes.

Reply:
Good idea!